toolchain/sem_ir/typed_insts.h

```diff
@@ -989,7 +1011,10 @@ struct NameRef {
 struct Namespace {
   static constexpr auto Kind =
       InstKind::Namespace.Define<Parse::AnyNamespaceId>(
-          {.ir_name = "namespace", .constant_kind = InstConstantKind::Always});
+          {.ir_name = "namespace",
+           // TODO: Modeling namespaces as unique doesn't properly handle
+           // namespace redeclarations.
```

Comment:
Why not use `Always` then?

Reply:
To preserve the existing (incorrect) behavior. I think switching to `Always` would do the right thing (we'd canonicalize based on the `NameScopeId`).